### ISSUE TITLE
Fails to apply a patch when the app name contains a dot

### ISSUE THREAD

**Issue by warpech** [Palindrom/palindrom-polymer-client#46]
### Steps to reproduce

1. Clone KitchenSink `develop`
2. Go to the directory `src/KitchenSink` in the Windows consle
3. Start the app by `star --name=Kitchen.Sink bin\Debug\KitchenSink.exe`
4. Go to http://localhost:8080/KitchenSink/Breadcrumb
5. Click on the "Chart" link in the menu

Starcounter version 2.3.1.7211

### Expected result

The current page should change to the Chart page.

### Actual result

The correct patch comes from the server but the page is not changed at the client.

I suspect a problem in the function `translateJSONPointerToPolymerPath` in `palindrom-polymer-client` or actually a problem in Polymer (https://github.com/Polymer/polymer/issues/3675, https://github.com/Polymer/polymer/issues/3127)

**Comment by alemoi** [comment]
@alshakero is it much to do, you think? Thanks!

**Comment by alshakero** [comment]
@alemoi I was on vacation last week. Tomorrow I'll look into it. And anytime!

**Comment by alshakero** [comment]
I can't reproduce this. `star.exe` seems to be trimming the name to `Kitchen` 
```sh
C:\Work\Starcounter\KitchenSink\src\KitchenSink>star --name=Kitchen.Sink bin\Debug\KitchenSink.exe
Kitchen -> default (started, default port 8080, admin 8181)                                              
```

I tried with `Kitchen\.Sink`, `'Kitchen.Sink'`. Neither worked.

**Comment by alemoi** [comment]
> I can't reproduce this. star.exe seems to be trimming the name to Kitchen

Yes, because its for Starcounter version 2.3.1.7211
I will send it to you by slack then.

**Comment by alshakero** [comment]
I tried this version and didn't work either 
![image](https://user-images.githubusercontent.com/17054134/30806974-2ff1985e-a1f8-11e7-9b39-66d757a372be.png)

**Comment by alemoi** [comment]
The important is step 3 in the @warpech description.

**Comment by alshakero** [comment]
Yes I understand, but as you see, I'm passing `--name=Kitchen.Sink` and it's being trimmed to `Kitchen`.

**Comment by alshakero** [comment]
@alemoi helped me reproduce it on Slack.

**Comment by alshakero** [comment]
Update: it's a Polymer issue and I'll report it to them today with a JSBin.

**Comment by warpech** [comment]
@alemoi most probably this will take months to fix in Polymer. Can you please escape the `.` to `_` in the JSON namespaces?

**Comment by alemoi** [comment]
Sure, i can do that @warpech!
So it will be "myapp_cs", "myapp_blend_cs", "myapp_exe", etc.
Are you sure its only about dots, or some other symbols can be deadly?

**Comment by warpech** [comment]
Polymer issues in the OT say that it is `.` and `#` which are not handled correctly by Polymer.

**Comment by alemoi** [comment]
Here is a new build for you guys to test: http://downloads.starcounter.com/download/Starcounter/Custom/2.3.1.7313
Let me know @alshakero if everything works well now :) Thanks!

**Comment by alshakero** [comment]
Everything seems to work. Should I bother with Polymer?

**Comment by alemoi** [comment]
Thanks @alshakero !
Are there any other objection @warpech of me NOT pushing the full app name fix to Starcounter? Any other apps that can suffer? Thanks!

**Comment by alemoi** [comment]
Pushed to the 2.3 and develop. Let's see if something bad pops up.

**Comment by alemoi** [comment]
I didn't write anything in changelog about this, because its just an intermediate change. Waiting for Polymer fixes.

**Comment by warpech** [comment]
@alemoi this is a big change. There might be apps that hardcoded the JSON namespace even though it was discouraged. Please write about it in the changelog.

**Comment by alemoi** [comment]
Sure thing @warpech Already pushed!

**Comment by warpech** [comment]
@alemoi I propose escaping `.` only in JSON namespaces but it seems that you escaped them generally for AppName? I think that might be confusing.

![2017-09-29_17-06-55](https://user-images.githubusercontent.com/566463/31022365-dcdbe568-a538-11e7-90c4-9f9a9adedd37.png)

**Comment by alemoi** [comment]
Yes, its basically now an `StarcounterEnvironment.AppName` that contains underscores, everything else are derivatives.

Do you want to keep `StarcounterEnvironment.AppName` with dots and escape in Json? I'm not sure how much Json relies on the `StarcounterEnvironment.AppName` better to ask @chrhol (he is not in this repo by the way, I suggest adding him here) I just made it safe, so its reflected in Json as well.

> @alemoi I propose escaping . only in JSON namespaces but it seems that you escaped them generally for AppName? I think that might be confusing.

Yeah, underscore is kind of confusing, but still its clear what is actually running. Have you found any functionality glitches?

**Comment by warpech** [comment]
Changing the status of this issue to "Frozen" as it is blocked by Polymer (Polymer/polymer#3675, Polymer/polymer#3127)
